Sources/HipMobileUI.Droid/Contracts/DroidNotificationPlayer.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (C) 2017 History in Paderborn App - Universität Paderborn
+//  
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//  
+//      http://www.apache.org/licenses/LICENSE-2.0
+//  
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+using Android.App;
+using Android.Content;
+using Android.Graphics;
+using Android.Media;
+using Android.OS;
+using Android.Support.V4.App;
+using PaderbornUniversity.SILab.Hip.Mobile.UI.NotificationPlayer;
+using Plugin.CurrentActivity;
+
+namespace PaderbornUniversity.SILab.Hip.Mobile.Droid.Contracts
+{
+    public class DroidNotificationPlayer : INotificationPlayer
+    {
+        // Use different IDs for different categories of notifications since notifications with the same ID override each other
+        private const int ExhibitNearbyNotificationId = 30052010;
+        private const int DefaultNotificationId = 06031993;
+
+        private const int InfoIcon = Resource.Drawable.info;
+
+        public void DisplayExhibitNearbyNotification(string title, string text, byte[] data = null)
+        {
+            DisplayDefaultNotification(title, text, ExhibitNearbyNotificationId, data);
+        }
+
+        /// <summary>
+        /// Display a simple notification with an optional large image
+        /// </summary>
+        /// <param name="title">Heading for the notification</param>
+        /// <param name="text">The message below the title</param>
+        /// <param name="id">Unique id for the notification</param>
+        /// <param name="bmpData">Data for the large image</param>
+        public void DisplayDefaultNotification(string title, string text, IComparable id = null, byte[] bmpData = null)
+        {
+            if (id == null)
+                id = DefaultNotificationId;
+
+            var mainActivity = (MainActivity)CrossCurrentActivity.Current.Activity;
```

Comment:
That's a weird Xamarin plugin. The concept of a single currently running `Activity` doesn't exist on Android, especially since Android 7.0 introduced [split-screen](https://developer.android.com/guide/topics/ui/multi-window.html) functionality.  Also, casting it `MainActivity` may break in the future, if we ever decide to introduce an additional `Activity`. Is there a good reason to not simply use the singleton `ApplicationContext` available using `Android.App.Application.Context`? Here's a good [reference](https://possiblemobile.com/2013/06/context/) to understand the differences between the various `Context` types.

Reply:
Found this way of getting the current context in another file. I used it thinking the original creators knew what they were doing. If using ApplicationContext is better I will adapt the code. And thanks for the reference.